src/codegen.cpp

```diff
@@ -7060,6 +7123,9 @@ static jl_returninfo_t get_specsig_function(jl_codectx_t &ctx, Module *M, Value
         param.addAttribute(Attribute::NoAlias);
         param.addAttribute(Attribute::NoCapture);
         param.addAttribute(Attribute::NoUndef);
+        param.addAttribute(Attribute::NonNull);
+        param.addDereferenceableAttr(props.union_bytes);
+        param.addAlignmentAttr(props.union_align);
```

Comment:
Do we actually ensure this when making this sret alloca? Normally this is the upper bounds that are permitted to be assumed, not the actual values at runtime, but this might be a special case.

Reply:
This what we do when we build the function in https://github.com/JuliaLang/julia/blob/8b8da91ad7701c5401947a749399720f5aa51418/src/codegen.cpp#L7806

Reply:
Ah, fair. That is the only place it matters too. Elsewhere it just adds a little bit of extra overhead to the function declaration to track some extraneous attributes.

Reply:
Doesn't it potenially give more information with how much memory an sret a function has ,i t's likely that the information is there on the pointer now but with opaque pointers it might not be?